Clang6: Build fails on Ubuntu 18.04 LTS

Just installed Ubuntu 18.04 on a virtual machine and tried to build ChakraCore on it, got the following errors that caused the build to fail:

```
./build.sh -j=6 --embed-icu
```

```
/home/fatcerberus/src/ChakraCore/lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.ICU.cpp:26:38: error: comparison of two values with different enumeration types ('PlatformAgnostic::UnicodeText::NormalizationForm' and 'UNormalization2Mode') [-Werror,-Wenum-compare]
                NormalizationForm::C == UNORM2_COMPOSE &&
                ~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~
/home/fatcerberus/src/ChakraCore/lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.ICU.cpp:27:38: error: comparison of two values with different enumeration types ('PlatformAgnostic::UnicodeText::NormalizationForm' and 'UNormalization2Mode') [-Werror,-Wenum-compare]
                NormalizationForm::D == UNORM2_DECOMPOSE &&
                ~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~
2 errors generated.
lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/build.make:230: recipe for target 'lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/Common/UnicodeText.ICU.cpp.o' failed
make[2]: *** [lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/Common/UnicodeText.ICU.cpp.o] Error 1
CMakeFiles/Makefile2:1010: recipe for target 'lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/all' failed
make[1]: *** [lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/all] Error 2
```

Don't know if there are any more errors since these two caused the build to abort.

--- a/lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.ICU.cpp
+++ b/lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.ICU.cpp
@@ -23,18 +23,18 @@ namespace PlatformAgnostic
         static const UNormalizer2 *StaticUNormalizerFactory(NormalizationForm nf)
         {
             static_assert(
-                NormalizationForm::C == UNORM2_COMPOSE &&
-                NormalizationForm::D == UNORM2_DECOMPOSE &&
+                static_cast<int>(NormalizationForm::C) == static_cast<int>(UNORM2_COMPOSE) &&
+                static_cast<int>(NormalizationForm::D) == static_cast<int>(UNORM2_DECOMPOSE) &&
                 // Deciding between ICU_NORMALIZATION_NFC and ICU_NORMALIZATION_NFKC
                 // Depends on ::KC and ::KD being INT_MIN + their non-K forms
                 // We can't just make them negative of their non-K forms because UNORM2_COMPOSE == 0
-                NormalizationForm::KC == INT_MIN + UNORM2_COMPOSE &&
-                NormalizationForm::KD == INT_MIN + UNORM2_DECOMPOSE &&
+                static_cast<int>(NormalizationForm::KC) == INT_MIN + static_cast<int>(UNORM2_COMPOSE) &&
+                static_cast<int>(NormalizationForm::KD) == INT_MIN + static_cast<int>(UNORM2_DECOMPOSE) &&
                 (
-                    NormalizationForm::Other != NormalizationForm::C &&
-                    NormalizationForm::Other != NormalizationForm::D &&
-                    NormalizationForm::Other != NormalizationForm::KC &&
-                    NormalizationForm::Other != NormalizationForm::KD
+                    static_cast<int>(NormalizationForm::Other) != static_cast<int>(NormalizationForm::C) &&
+                    static_cast<int>(NormalizationForm::Other) != static_cast<int>(NormalizationForm::D) &&
+                    static_cast<int>(NormalizationForm::Other) != static_cast<int>(NormalizationForm::KC) &&
+                    static_cast<int>(NormalizationForm::Other) != static_cast<int>(NormalizationForm::KD)
                 ),
                 "Invalid NormalizationForm enum configuration"
             );

